lib/internal/streams/end-of-stream.js

```diff
@@ -28,40 +28,51 @@ function eos(stream, opts, callback) {
 
   callback = once(callback);
 
+  const onerror = (err) => {
+    callback.call(stream, err);
+  };
+
+  let writableFinished = stream.writableFinished ||
+    (stream._writableState && stream._writableState.finished);
+  let readableEnded = stream.readableEnded ||
+    (stream._readableState && stream._readableState.endEmitted);
+
+  if (writableFinished || readableEnded || stream.destroyed) {
+    if (opts.error !== false) stream.on('error', onerror);
+    // A destroy(err) call emits error in nextTick.
+    process.nextTick(callback.bind(stream));
```

Comment:
Wouldn't this call the cb with no error incase it's been destroyed with no error as well?
We'd normally call it with the premature error in that case.

Reply:
I'm not sure I follow. Do you mean it calls the callback with no error if it has been destroyed **with** an error?

Reply:
aah I think I understand, `callback` is wrapped in `once`, so no that should not be a problem.